### PR TITLE
fix(DB): Against the Legion requirements

### PR DESCRIPTION
**Pull request by ZhengPeiRu21**
## Changes Proposed:
- Fixes Against the Legion and related quests to appear correctly when requirements are met

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/15129
- Closes https://github.com/chromiecraft/chromiecraft/issues/5039

## SOURCE:

See original issue

## Tests Performed:

- Tested in game, works

## How to Test the Changes:

See original issue reports.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1677781820338823400.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1677781820338823400.sql:2]: Missing DELETE Here

Comment (ZhengPeiRu21) [reply data/sql/updates/pending_db_world/rev_1677781820338823400.sql]: Sorry, thanks for catching! Fixed.

**Comment by Maelthyrr** [comment]
Hi, thanks for the PR. Tested with Alliance and Horde character and work fine. After completing one of two quests, the tree others quests are available.

File: data/sql/updates/pending_db_world/rev_1677781820338823400.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
+--
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId` = 19 AND `SourceEntry` IN (10641, 10669);
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1677781820338823400.sql:2]:
```suggestion
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId` = 19 AND `SourceEntry` IN (10641, 10668, 10669);
```

**Review by Gultask — APPROVED** [review]

**Pull request merged**